### PR TITLE
[Executor] Record source url for image

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

Record source url for image.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/contracts/multimedia.py [hunk 1]
```diff
@@ -15,13 +15,14 @@ def __new__(cls, value: bytes, *args, **kwargs):
         # See https://docs.python.org/3/reference/datamodel.html#object.__new__
         return super().__new__(cls, value)
 
-    def __init__(self, value: bytes, mime_type: str):
+    def __init__(self, value: bytes, mime_type: str, sourceUrl: str = None):
```
Comment (thy09) [inline src/promptflow/promptflow/contracts/multimedia.py:18]: Should be Optional[str] = None

File: src/promptflow/promptflow/_utils/multimedia_utils.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def _create_image_from_url(url: str, mime_type: str = None):
         if not mime_type:
             format = imghdr.what(None, response.content)
             mime_type = f"image/{format}" if format else "image/*"
-        return Image(response.content, mime_type=mime_type)
+        return Image(response.content, mime_type=mime_type, sourceUrl=url)
```
Comment (melionel) [inline src/promptflow/promptflow/_utils/multimedia_utils.py:87]: sourceUrl -> source_url ?

File: src/promptflow/promptflow/contracts/multimedia.py (hunk 1 above)
Comment (lumoslnt) [reply src/promptflow/promptflow/contracts/multimedia.py]: Refined.

File: src/promptflow/promptflow/_utils/multimedia_utils.py (hunk 2 above)
Comment (lumoslnt) [reply src/promptflow/promptflow/_utils/multimedia_utils.py]: Refined.

**Review by thy09 — APPROVED** [review]

**Review by melionel — APPROVED** [review]

**Pull request merged**